### DISCUSSION TITLE
[SOLVED] Filter out updated articles

### DISCUSSION THREAD

**Discussion by Mariocraft95** [martinrotter/rssguard#1501] [Q&A] [answered]
Used a couple different RSS feeds, but this is the only one I have noticed this problem with. I appreciate the people who use their free time to work on this. 

I use RSS Guard as a Youtube feed to limit my time scrolling on Youtube, and I have a folder of creators that only upload a few times a year. Opened my feed to see multiple videos a day from them, but they were all old videos. Upon further investigation, it seems like RSS guard is adding things to my feed if the "updated" tag gets updated. For example, https://www.youtube.com/feeds/videos.xml?channel_id=UCK376qNDlNZZDNHsnaWuTeg (AntVenom youtuber). He apparently makes updates to old video relatively frequently. I was noticing that the old videos showing up in my feed were actually videos that were recently updated.

Is there any way to disable this? I have not had a RSS Reader do this by default. This unfortunately completely clutters my RSS feed.  Thank you,

**Comment by Ac314** [comment, marked as answer]
This is happening not because of the specific YouTube channel - this is the default YouTube behavior for all channels. They "promote" (meaning update the "Updated" tags in ATOM feeds) old videos periodically based of unknown algorithms. 

This is not noticeable if the specific channel is updated frequently by itself (new videos are published often). But for rarely updated channels you will observe such behavior constantly. 

YouTube ATOM feed for any channel is always 15 newest entries ("newest" means newest based on "Published" tags, not "Updated"). So oldest entries will be cleaned by YouTube from the feed eventually according to "Published" tag. But RSS Guard consider entries newer or older based on "Updated" tag (which is quite reasonable). Some old entries can be made "new" in RSS Guard based on "Updated" tags, so actually newer items will be deleted from the database, But because they are still present in YouTube feed they will be fetched again.

Possible ways to avoid this (maybe more exist so someone could add to this):
1) For YouTube channels in question keep more articles in database (Feed properties -> Articles -> Limiting amount of articles in feed -> In database, keep N articles). Set it to 30 and you will be OK.
2) Add the filter for such channels to completely avoid taking into consideration the "Updated" tag. See documentation for the filters (https://rssguard.readthedocs.io/en/stable/features/filters.html), it contains example of such filter (the title is "Use published element instead of updated element (for ATOM entries only)"). I do not know exactly if it still works or no, I did not try.

**Reply by Mariocraft95** [reply]
On pressing the test button on the recommended filter, I get 
"EXISTING articles filtering error: 'TypeError: Cannot read property '__text' of undefined'."

Though, that might just be due to the test being an RSS feed which does not have the updated tag. 
Testing on my feeds, seems to do the trick! Thank you so much!
